### ISSUE TITLE
No GPU and CPU related result showed in Profile webpage after following guide 

### ISSUE THREAD

**Issue by quxinqq** [tensorflow/profiler#347]
My Enviroment:
Linux version 3.10.0-957.27.2.el7.x86_64 (mockbuild@kbuilder.bsys.centos.org) (gcc version 4.8.5 20150623 (Red Hat 4.8.5-36)
tensorboard==2.2.2
tensorboard-plugin-profile==2.4.0
tensorflow==2.2.0
conda-build version : 3.15.1
python version : 3.7.0
cuda version:
![图片](https://user-images.githubusercontent.com/88089887/127318492-e855c32b-d816-4d93-b9a8-2e90a4e30acf.png)
cudnn version:
![图片](https://user-images.githubusercontent.com/88089887/127319050-9ba1a456-d679-4f71-902d-3231ca53e37b.png)

Profile guide link:
https://www.tensorflow.org/tensorboard/tensorboard_profiling_keras

My Code:
from datetime import datetime
from packaging import version
import tensorflow_datasets as tfds
import tensorflow as tf
if __name__ == '__main__':   
    gpus = tf.config.experimental.list_physical_devices(device_type='GPU')
    tf.config.experimental.set_visible_devices(devices=gpus[0], device_type='GPU') 
    tfds.disable_progress_bar()
    (ds_train, ds_test), ds_info = tfds.load(
        'mnist',
        split=['train', 'test'],
        shuffle_files=True,
        as_supervised=True,
        with_info=True,
    )
    def normalize_img(image, label):
      """Normalizes images: `uint8` -> `float32`."""
      return tf.cast(image, tf.float32) / 255., label

    ds_train = ds_train.map(normalize_img)
    ds_train = ds_train.batch(128)
    ds_test = ds_test.map(normalize_img)
    ds_test = ds_test.batch(128)

    model = tf.keras.models.Sequential([
      tf.keras.layers.Flatten(input_shape=(28, 28, 1)),
      tf.keras.layers.Dense(128,activation='relu'),
      tf.keras.layers.Dense(10, activation='softmax')
    ])
    model.compile(
        loss='sparse_categorical_crossentropy',
        optimizer=tf.keras.optimizers.Adam(0.001),
        metrics=['accuracy']
    )
    # Create a TensorBoard callback
    logs = os.path.join(
        "logs",
        "fit",
        datetime.now().strftime("%Y%m%d-%H%M%S"),
    )
    tboard_callback = tf.keras.callbacks.TensorBoard(log_dir = logs,
                                                     histogram_freq = 1,
                                                     profile_batch = '500,520')

    model.fit(ds_train,
              epochs=3,
              validation_data=ds_test,
              callbacks = [tboard_callback])
![图片](https://user-images.githubusercontent.com/88089887/127317530-c41b8d9e-ef17-4cc3-88c3-46e284842470.png)

![Issue_Pofile_Picture](https://user-images.githubusercontent.com/88089887/127317160-a4ecddaa-9c4d-47b4-b051-4e78b19cd29e.jpg)
![图片](https://user-images.githubusercontent.com/88089887/127317932-883df55a-7e13-4618-b41f-841f469be289.png)
![图片](https://user-images.githubusercontent.com/88089887/127318074-4b76172b-61b3-4fb1-b58f-e7b30d96b34a.png)

**Comment by yisitu** [comment]
Hi, it seems that this us using TF 2.4 with CUDA 10.1. Would it be possible to use one of the supported configuration of TensorFlow X CUDA?

https://www.tensorflow.org/install/source#linux

**Comment by yisitu** [comment]
The guide seems to be using TensorFlow 2.2, the same configuration which you have used. Would it be possible for you to use the latest configuration from here: https://www.tensorflow.org/install/source#gpu



Version | Python version | Compiler | Build tools | cuDNN | CUDA
-- | -- | -- | -- | -- | --
tensorflow-2.5.0 | 3.6-3.9 | GCC 7.3.1 | Bazel 3.7.2 | 8.1 | 11.2

**Comment by yisitu** [comment]
Here is a colab which I had used from https://www.tensorflow.org/tensorboard/tensorboard_profiling_keras:

https://colab.research.google.com/github/tensorflow/tensorboard/blob/master/docs/tensorboard_profiling_keras.ipynb

I managed to get a profile after adding `--load_fast=false` to tensorboard.
```diff
-%tensorboard --logdir=logs
+%tensorboard --logdir=logs --load_fast=false
```


![AELBaD243CPvEMM](https://user-images.githubusercontent.com/60493024/127408513-05b96055-3d85-4d4a-ace6-8a592f79bbba.png)

**Comment by yisitu** [comment]
I've also released a new tensorboard_plugin_profile=2.5.0 that fixes this.

**Comment by quxinqq** [comment]
Hi Yisitu

    Thanks very much for your answers and suggestion.I have already fixed this issue without changing or updating versions of Tensorflow , CUDA&Driver, cudnn...
    Root cause is: Nvidia driver is limiting non-administrator user access profile performance counter.
    The effective method is changing from normal user to root user in Linux and then run python code.
    The error log infor:
2021-07-30 12:30:28.946605: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1247] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 30233 MB memory) -> physical GPU (device: 0, name: Tesla V100-PCIE-32GB, pci bus id: 0000:b3:00.0, compute capability: 7.0)
2021-07-30 12:30:30.278858: I tensorflow/core/profiler/lib/profiler_session.cc:159] Profiler session started.
2021-07-30 12:30:30.278940: I tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1363] Profiler found 1 GPUs
2021-07-30 12:30:30.279184: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libcupti.so.10.1'; dlerror: libcupti.so.10.1: cannot open shared object file: No such file or directory
2021-07-30 12:30:30.279220: E tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1408] function cupti_interface_->Subscribe( &subscriber_, (CUpti_CallbackFunc)ApiCallback, this)failed with error CUPTI could not be loaded or symbol could not be found.
2021-07-30 12:30:30.279247: E tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1447] function cupti_interface_->ActivityRegisterCallbacks( AllocCuptiActivityBuffer, FreeCuptiActivityBuffer)failed with error CUPTI could not be loaded or symbol could not be found.
2021-07-30 12:30:30.279282: E tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1430] function cupti_interface_->EnableCallback( 0 , subscriber_, CUPTI_CB_DOMAIN_DRIVER_API, cbid)failed with error CUPTI could not be loaded or symbol could not be found.


![图片](https://user-images.githubusercontent.com/88089887/127621412-5c012026-19f3-462e-85ac-502c2d094742.png)
